File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minSoFar = prices[0];
        int maxProfit = INT_MIN;
        int profit;
        for(auto price: prices){
            minSoFar = min(minSoFar, price);
            profit = price-minSoFar;
            maxProfit = max(maxProfit, profit);
        }
        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_so_far = prices[0]
        max_profit = 0

        for i, val in enumerate(prices):
            min_so_far = min(min_so_far, val)
            max_profit = max(max_profit, val - min_so_far)

        return max_profit
        